server/src/main/java/org/elasticsearch/search/vectors/AbstractIVFKnnVectorQuery.java

```diff
@@ -118,9 +131,56 @@ public Query rewrite(IndexSearcher indexSearcher) throws IOException {
         KnnCollectorManager knnCollectorManager = getKnnCollectorManager(numCands, indexSearcher);
         TaskExecutor taskExecutor = indexSearcher.getTaskExecutor();
         List<LeafReaderContext> leafReaderContexts = reader.leaves();
-        List<Callable<TopDocs>> tasks = new ArrayList<>(leafReaderContexts.size());
-        for (LeafReaderContext context : leafReaderContexts) {
-            tasks.add(() -> searchLeaf(context, filterWeight, knnCollectorManager));
+
+        int totalBudget = (int) (reader.numDocs() * visitedRatio);
```

Comment:
we can do it this way. We should iterate the leaves, and gather the vector counts directly. Its possible that there are many docs without vectors.